src/Interfaces/Requests/UpdateSynonymRuleAdaptor.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace SilverStripe\Forager\Interfaces\Requests;
+
+use SilverStripe\Forager\Service\Query\SynonymRule;
+
+interface UpdateSynonymRuleAdaptor
+{
+
+    /**
+     * @return string|int the ID of the synonym rule that was updated
+     */
+    public function process(
+        string|int $synonymCollectionId,
+        string|int $synonymRuleId,
+        SynonymRule $synonymRule
+    ): string|int;
```

Comment:
wondering if update should return all the rule data. Just thinking of occasions where a slightly different format could come back (truncation, order etc). Similar to create - thoughts?

Reply:
This would be more consistent with Algolia, Typsense, and App search. Elasticsearch doesn't provide such a response, but I think that's because it's a cluster level change, and so it's response is focused on giving you feedback about that change; I don't think that's [this module's problem] though - seems like most services give you this sort of feedback.

Reply:
maybe it could return an object that has _at least_ an id - and possibly more data. this would cover both types of response?

Reply:
`Results\SynonymRule` covers that case - only the `id` field is mandatory when populating that object.